lib/mail/version_specific/ruby_1_9.rb

```diff
@@ -125,7 +125,11 @@ def Ruby19.q_value_decode(str)
     def Ruby19.param_decode(str, encoding)
       str = uri_parser.unescape(str)
       str = charset_encoder.encode(str, encoding) if encoding
-      str
+      decoded = str.encode(Encoding::UTF_8, :invalid => :replace, :replace => "")
+      decoded.valid_encoding? ? decoded : decoded.encode(Encoding::UTF_16LE, :invalid => :replace, :replace => "").encode(Encoding::UTF_8)
+    rescue Encoding::UndefinedConversionError, ArgumentError, Encoding::ConverterNotFoundError
+      warn "Encoding conversion failed #{$!}"
```

Comment:
Yes, yes it will. And it annoys me every time it shows up in my tests. Should this instead add to Mail.errors or something?

Do you want me to do this before or after refactoring the conversion to UTF-8 the other _decode methods use